File: 158.两个字符串是变位词/Solution.py
class Solution:
    """
    @param s: The first string
    @param b: The second string
    @return true or false
    """
    def anagram(self, s, t):
        # write your code here
        n = len(s)
        if len(t) != n:
            return False
        d = {}
        for v in s:
            if d.get(v, -1) == -1:
                d[v] = 1
            else:
                d[v] += 1
        for v in t:
            if d.get(v, -1) == -1:
                return False
            d[v] -= 1
            if d[v] < 0:
                return False
        return True
        

File: 158.两个字符串是变位词/Solution.cpp
class Solution {
public:
    /**
     * @param s: The first string
     * @param b: The second string
     * @return true or false
     */
    bool anagram(string s, string t) {
        // write your code here
        int lens = s.size(), lent = t.size();
		if(lens != lent)
			return false;
		vector<int> count(128,0);
		for (string::iterator it=s.begin(); it!=s.end(); it++)
			count[*it]++;
		for(string::iterator it=t.begin(); it!=t.end(); it++)
			if(--count[*it] < 0)
				return false;
		return true;
    }
};